core/src/services/aliyun_drive/core.rs

```diff
@@ -0,0 +1,506 @@
+            "size": size,
+            "pre_hash": pre_hash,
+            "content_hash": content_hash,
+            "content_hash_name": content_hash.is_some().then_some("sha1"),
+            "proof_code": proof_code,
+            "proof_version": proof_code.is_some().then_some("v1"),
+        }))
+        .map_err(new_json_serialize_error)?;
+        let req = Request::post(format!("{}/adrive/v1.0/openFile/create", self.endpoint))
+            .body(Buffer::from(Bytes::from(body)))
+            .map_err(new_request_build_error)?;
+        self.send(req, token.as_deref()).await
+    }
+
+    pub async fn create(
+        &self,
+        parent_file_id: Option<&str>,
+        name: &str,
+        typ: CreateType,
+        check_name_mode: CheckNameMode,
+    ) -> Result<Buffer> {
+        self.create_with_rapid_upload(parent_file_id, name, typ, check_name_mode, None, None)
+            .await
+    }
+
+    pub async fn get_download_url(&self, file_id: &str) -> Result<String> {
+        let (token, drive_id) = self.get_token_and_drive().await?;
+        let body = serde_json::to_vec(&json!({
+            "drive_id": &drive_id,
+            "file_id": file_id,
+        }))
+        .map_err(new_json_serialize_error)?;
+        let req = Request::post(format!(
+            "{}/adrive/v1.0/openFile/getDownloadUrl",
+            self.endpoint
+        ))
+        .body(Buffer::from(Bytes::from(body)))
+        .map_err(new_request_build_error)?;
+        let res = self.send(req, token.as_deref()).await?;
+        let output: GetDownloadUrlResponse =
+            serde_json::from_reader(res.reader()).map_err(new_json_serialize_error)?;
+        Ok(output.url)
+    }
+
+    pub async fn move_path(&self, file_id: &str, to_parent_file_id: &str) -> Result<()> {
+        let (token, drive_id) = self.get_token_and_drive().await?;
+        let body = serde_json::to_vec(&json!({
+            "drive_id": &drive_id,
+            "file_id": file_id,
+            "to_parent_file_id": to_parent_file_id,
+            "check_name_mode": CheckNameMode::AutoRename,
+        }))
+        .map_err(new_json_serialize_error)?;
+        let req = Request::post(format!("{}/adrive/v1.0/openFile/move", self.endpoint))
+            .body(Buffer::from(Bytes::from(body)))
+            .map_err(new_request_build_error)?;
+        self.send(req, token.as_deref()).await?;
+        Ok(())
+    }
+
+    pub async fn update_path(&self, file_id: &str, name: &str) -> Result<()> {
+        let (token, drive_id) = self.get_token_and_drive().await?;
+        let body = serde_json::to_vec(&json!({
+            "drive_id": &drive_id,
+            "file_id": file_id,
+            "name": name,
+            "check_name_mode": CheckNameMode::Refuse,
+        }))
+        .map_err(new_json_serialize_error)?;
+        let req = Request::post(format!("{}/adrive/v1.0/openFile/update", self.endpoint))
+            .body(Buffer::from(Bytes::from(body)))
```

Comment:
Buffer can be built from `Vec<u8>` directly, so we don't need to build `Bytes` first.